tests/runtime/model_management/test_model_manager.py

```diff
@@ -773,6 +774,164 @@ def test_lazy_load_ephemeral_model():
             assert model_id in manager.loaded_models
 
 
+def test_deploy_undeploy_model(deploy_good_model_files):
+    """Test that a model can be deployed by copying to the local models dir"""
+    with TemporaryDirectory() as cache_dir:
+        with non_singleton_model_managers(
+            1,
+            {
+                "runtime": {
+                    "local_models_dir": cache_dir,
+                    "lazy_load_local_models": True,
+                    "lazy_load_poll_period_seconds": 0,
+                },
+            },
+            "merge",
+        ) as managers:
+            manager = managers[0]
+            model_name = "my-model"
+
+            # Make sure model is not currently loaded
+            with pytest.raises(CaikitRuntimeException) as excinfo:
+                manager.retrieve_model(model_name)
+                assert excinfo.value.status_code == grpc.StatusCode.NOT_FOUND
+
+            # Do the deploy (pass wait through to load)
+            loaded_model = manager.deploy_model(
+                model_name, deploy_good_model_files, wait=True
+            )
+            assert loaded_model
+            assert loaded_model.loaded
+
+            # Make sure model can be retrieved and exists in the local models dir
+            assert manager.retrieve_model(model_name)
+            assert os.path.isdir(os.path.join(cache_dir, model_name))
+
+            # Make sure model cannot be deployed over
+            with pytest.raises(CaikitRuntimeException) as excinfo:
+                manager.deploy_model(model_name, deploy_good_model_files)
+                assert excinfo.value.status_code == grpc.StatusCode.ALREADY_EXISTS
+
+            # Undeploy the model
+            manager.undeploy_model(model_name)
+
+            # Make sure the model is not loaded anymore and was removed from
+            # local models dir
+            with pytest.raises(CaikitRuntimeException) as excinfo:
+                manager.retrieve_model(model_name)
+                assert excinfo.value.status_code == grpc.StatusCode.NOT_FOUND
+            assert not os.path.exists(os.path.join(cache_dir, model_name))
+
+
+@pytest.mark.parametrize(
+    "invalid_fname",
+    ["", "\t\n  ", "/foo/bar.txt"],
```

Comment:
nit: can we add an expected reason match as well for these? Like `".*Cannot use absolute paths for model files.*"`

and then `with pytest.raises(CaikitRuntimeException, match=expected_reason)`

Reply:
Good call